lib/puppet/provider/package/npm.rb

```diff
@@ -51,8 +51,8 @@ def query
   end
 
   def latest
-    if /#{resource[:name]}@([\d\.]+)/ =~ npm('outdated', '--global',  resource[:name])
-      @latest = $1
+    if /#{resource[:name]}@([\d\.]+)/ =~ npm('outdated', '--global', resource[:name])
+      @latest = Regexp.last_match(1)
     else
       @property_hash[:ensure] unless @property_hash[:ensure].is_a? Symbol
     end
```

Comment:
this whole function feels very unruby… not sure ow to make it better
or simplify it for that matter
perhaps [split](http://ruby-doc.org/core-2.3.0/String.html#method-i-split) is all we need

so, basically,
```ruby
def latest
  outdated = npm('outdated', '--global', resource[:name]).split('@', 1) # no idea what i'm doing
  return outdated if outdated =~ /^(\d\.)+$/ # n.b.: this doesn't match builds, alpha's rcs, etc…
  return @property_hash[:ensure] unless @property_hash[:ensure].is_a? Symbol # why? we could just to_s
  # cuz otherwise this returns nil
end
```